### PR TITLE
Don't delete node pools for imported clusters

### PR DESCRIPTION
**Pull request by cmurphy**
Also fix examples for imported clusters.

### CODE REVIEW

File: controller/aks-cluster-config-handler.go [hunk 1]
```diff
@@ -627,53 +627,55 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, secretsCache w
 		return config, err
 	}
 
-	downstreamNodePools, err := utils.BuildNodePoolMap(config.Spec.NodePools, config.Spec.ClusterName)
-	if err != nil {
-		return config, err
-	}
+	if config.Spec.NodePools != nil {
```
Comment (thedadams) [inline controller/aks-cluster-config-handler.go:630]: nit: put the `aks.NewAgentPoolClient` call above in this if-block because it isn't needed otherwise.

**Review by thedadams — DISMISSED** [review]

**Review by thedadams — APPROVED** [review]

**Review by rmweir — COMMENTED** [review]
To clarify this is more, don't delete node pools if nodepools field is not managed, correct? So if you do have an imported cluster but you've edited the nodepools then upstream node pool not included in spec will be deleted? If this all is correct then LGTM.

**Comment by cmurphy** [comment]
> To clarify this is more, don't delete node pools if nodepools field is not managed, correct?

Yes, that's correct.

**Review by cbron — APPROVED** [review]

**Comment by cbron** [comment]
@cmurphy can you add an issue to this PR

**Comment by cmurphy** [comment]
Issue: https://github.com/rancher/rancher/issues/32734

**Pull request merged**